### PR TITLE
patch libtommath for CVE-2023-36328

### PR DESCRIPTION
**Pull request by bfjelds**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [ ] The toolchain has been rebuilt successfully (or no changes were made to it)
- [ ] The toolchain/worker package manifests are up-to-date
- [ ] Any updated packages successfully build (or no packages were changed)
- [ ] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [ ] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [ ] All package sources are available
- [ ] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [ ] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [ ] All source files have up-to-date hashes in the `*.signatures.json` files
- [ ] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [ ] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 
Adapt nist linked patch to libtommath 1.1.0 to address CVE-2023-36328

###### Change Log  
- CVE-2023-36328

###### Does this affect the toolchain?  
**NO**

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2023-36328

###### Test Methodology

- https://dev.azure.com/mariner-org/mariner/_build/results?buildId=418762&view=results

### CODE REVIEW

File: SPECS-EXTENDED/libtommath/libtommath.spec [hunk 1]
```diff
@@ -26,6 +27,7 @@ applications that use %{name}.
 
 %prep
 %setup -q
+%patch0 -p1
 # Fix permissions on installed library
 sed -i -e 's/644 $(LIBNAME)/755 $(LIBNAME)/g' makefile.shared
 # Fix pkgconfig path
```
Comment (bfjelds) [inline SPECS-EXTENDED/libtommath/libtommath.spec:33]: changelog entry

Comment (bfjelds) [reply SPECS-EXTENDED/libtommath/libtommath.spec:33]: new version

File: SPECS-EXTENDED/libtommath/libtommath.spec [hunk 2]
```diff
@@ -26,6 +27,7 @@ applications that use %{name}.
 
 %prep
 %setup -q
+%patch0 -p1
```
Comment (mfrw) [inline SPECS-EXTENDED/libtommath/libtommath.spec:30]: Consider:
```suggestion
%autosetup -p1
```

**Review by mfrw — APPROVED** [review]
LGTM, with a suggestion.

**Pull request merged**